bazel/repository_locations.bzl

```diff
@@ -1,4 +1,16 @@
 # This should match the schema defined in external_deps.bzl.
+
+PROTOC_VERSIONS = dict(
```

Comment:
@moderation im guessing its ok for `repository_locations.bzl` to share `locals()` namepsace with `REPOSITORY_LOCATIONS_SPEC` 

it was kinda necessary to do it this way - rather than post-processing for a few reasons, but essentially becaue it does introduce these new "deps" albeit from an existing release

we might also want to consider adding a `toolchain_downloads` or similar to the repo locations spec to automate this for other deps

Reply:
re this - this also raises a load of extra dependency issues in the dep checker - there are a bunch of ways to fix/workaround but im inclined not to as it provides (~correct) checking on the deps added here